### PR TITLE
Fix version specifiers of python packages for conda

### PR DESCRIPTION
**Pull request by tobiasdiez**
### 📚 Description

For some python packages, conda installed an incompatible version which was then uninstalled and replaced by the correct version during the `pip install src` step. This PR adds the missing version specifiers in the conda dist files. It also adds a check in the conda ci workflow and runs the conda workflow automatically on package updates (using the label) to prevent such mismatches in the future.

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ ] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by tobiasdiez** [comment]
Test run: https://github.com/sagemath/sage/actions/runs/4523467344/jobs/7966614875

File: .github/workflows/ci-conda.yml [hunk 1]
```diff
@@ -19,6 +22,12 @@ jobs:
     name: Conda
     runs-on: ${{ matrix.os }}
     
+    if: |
+      github.event.action != 'labeled' ||
```
Comment (mkoeppe) [inline .github/workflows/ci-conda.yml:27]: Could you add a comment that explains what this condition is intended to do?

Comment (tobiasdiez) [reply .github/workflows/ci-conda.yml:27]: Done.

Comment (mkoeppe) [reply .github/workflows/ci-conda.yml:27]: Thanks, but why should it NOT run when the action is "labeled"?

Comment (tobiasdiez) [reply .github/workflows/ci-conda.yml:27]: When the action is "labeled" it should only run when one of the specified labels is added. So in this case this resolves to `false || true`.

Comment (mkoeppe) [reply .github/workflows/ci-conda.yml:27]: Oh, now I see. Thanks.

**Review by mkoeppe — APPROVED** [review]

**Comment by tobiasdiez** [comment]
Thanks for the review!

**Pull request merged**